### PR TITLE
chore: make CommandHandler.handle return a Result

### LINKED ISSUE #2340: `CommandHandler` does not report validation errors to `CommandRunner`
# Feature Request

Currently some `CommandHandler`s intentionally throw exceptions to notify validation errors, that are catched by the `CommandRunner`. This makes the error detection harder, because all the errors are seen as unexpected.

## Which Areas Would Be Affected?
command handlers/runners

## Why Is the Feature Desired?
improve error reporting

## Solution Proposal
make `CommandHandler.handle` return a `Result` and make the `CommandRunner` evaluate it, the result would then be returned to the caller (`CommandProcessor`) and logged accordingly.

### PR DESCRIPTION
**Pull request by majadlymhmd**
## What this PR changes/adds

the CommandHandler.handle returns a Result instead of void
## Why it does that

to make it easier to detect the errors and improve the reposrting of them
## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Closes #2340 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-edc/Connector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

**Comment by majadlymhmd** [comment]
@ndr-brt could you please review it

**Comment by ndr-brt** [comment]
@majadlymhmd no need to send me a message, just add me as a reviewer (top right corner) ;)

File: spi/common/core-spi/src/main/java/org/eclipse/edc/spi/command/CommandHandler.java [hunk 1]
```diff
@@ -31,6 +33,6 @@
     /**
      * Processes the command.
      */
-    void handle(T command);
+    Result<Void> handle(T command);
```
Comment (ndr-brt) [inline spi/common/core-spi/src/main/java/org/eclipse/edc/spi/command/CommandHandler.java:36]: Now that the `CommandHandler` returns a `Result`, the `CommandRunner` can return it's result (as specified in the issue #2340 )

**Review by ndr-brt — CHANGES_REQUESTED** [review]

**Comment by paullatzelsperger** [comment]
@majadlymhmd since you are touching core elements of the command processing subsystem, please provide a written decision record that outlines the impact and your proposed changes as per our contribution guidelines.

As a small preview: I don't think you can complete this task without touching the `CommandRUnner`.

**Pull request closed without merging**